node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -206,7 +210,8 @@ case class DataMessageHandler(
         logger.info(
           s"Received block message with hash ${msg.block.blockHeader.hash.flip}")
         Future {
-          callbacks.onBlockReceived.foreach(_.apply(msg.block))
+          callbacks.onBlockReceived.foldLeft(FutureUtil.unit)((acc, callback) =>
```

Comment:
No need to wrap it in `Future` here, `foldLeft` returns a `Future` now.
```suggestion
          callbacks.onBlockReceived.foldLeft(FutureUtil.unit)((acc, callback) => acc.flatMap(_ => callback(msg.block))).map(_ => this)
```